From 95859599d56627495ad71a14e9573dc17cf5584f Mon Sep 17 00:00:00 2001
From: Mario Limonciello <mario.limonciello@amd.com>
Date: Fri, 11 Aug 2023 16:35:01 -0500
Subject: [PATCH 9/9] cpupower: Fix querying boost support for amd-pstate-epp
 mode

When running in active mode the driver for amd-pstate has -epp appended.
This throws off amd-pstate detection.

To detect amd-pstate look for the prefix instead.

Fixes: ffa5096a7c33 ("cpufreq: amd-pstate: implement Pstate EPP support for the AMD processors")
Reported-by: chenhq2005@gmail.com
Closes: https://bugzilla.kernel.org/show_bug.cgi?id=217755
Signed-off-by: Mario Limonciello <mario.limonciello@amd.com>
---
 tools/power/cpupower/utils/helpers/misc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/power/cpupower/utils/helpers/misc.c b/tools/power/cpupower/utils/helpers/misc.c
index bfdd9273b..d7883204b 100644
--- a/tools/power/cpupower/utils/helpers/misc.c
+++ b/tools/power/cpupower/utils/helpers/misc.c
@@ -150,7 +150,7 @@ bool cpupower_amd_pstate_enabled(void)
 	if (!driver)
 		return ret;
 
-	if (!strcmp(driver, "amd-pstate"))
+	if (!strncmp(driver, "amd-pstate", 10))
 		ret = true;
 
 	cpufreq_put_driver(driver);
-- 
2.42.0.rc0.25.ga82fb66fed

